sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1328,6 +1330,8 @@ private RxDocumentServiceRequest addBatchHeaders(RxDocumentServiceRequest reques
 
             request.setPartitionKeyInternal(partitionKeyInternal);
             request.getHeaders().put(HttpConstants.HttpHeaders.PARTITION_KEY, Utils.escapeNonAscii(partitionKeyInternal.toJson()));
+        } else if(serverBatchRequest instanceof PartitionKeyRangeServerBatchRequest) {
+            request.setPartitionKeyRangeIdentity(new PartitionKeyRangeIdentity(((PartitionKeyRangeServerBatchRequest) serverBatchRequest).getPartitionKeyRangeId()));
```

Comment:
now that we are setting partitionKeyRangeId explicitly how do we support partition split scenario?

If after populating pkrangeId here partition gets splitted how do we react? what would be the the sequence of actions on partition split?

Reply:
Right now we will just return a gone and we can expect user to retry. To support that retry we have to move sink before grouping operations or keep two sinks for different retries.

Reply:
I think we need to provide that retry-logic out-of-the-box (vs. expecting users to do it)

Reply:
The older bulk executor has support for partition splits... we should try to get at least the same with the new implementation. 
I think this whole retry logic might need some careful thought; ideally for any CosmosItemOperation that needs to be retried, it gets added back into the producer (Flux) after some initial logic that will take care of delays or handling partition splitting...

Reply:
Do we get any deals information for split case? If yes we can add logic for it, otherwise we have retry for individual operations and does it only once.

Reply:
Yes, there's a substatus code that indicates the failure is caused by a split; see https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/PartitionProcessorImpl.java#L128

Reply:
Yeah, so we are using it to put the operation in correct sink or if retry option is exhausted, return the exception.

Reply:
physical partitioning of data and partition split are internal details of our service implementation. We should not expose this to the end user as contract. 

The sdk needs to retry on partition spit internally. we do this for other operations today. 

This can go to a different PR, but without it we can't announce bulk as feature.

Reply:
We are not exposing it as a contract, and we are retrying.